File: sorting_python/bubblesort.py
from sort import Sort

class BubbleSort(Sort):

    def bubblesort(self,arr,low,high):
        for i in range(low,high):
            for j in range(low,high-i):
                if arr[j][0] > arr[j+1][0]:
                   arr[j],arr[j+1]=arr[j+1],arr[j]

    def sort_data(self):
        data=self.getData()
        self.bubblesort(data,0,len(data)-1)

File: sorting_cpp/src/bubble_sort.cpp
#include<iostream>
#include "../include/bubble_sort.hpp"
using namespace std;
template class BubbleSort<string>;
template<class T>
void BubbleSort<T>::bubble_sort(int low,int high)
{
        int i,j;
        for(i=low;i<high;i++)
        {
            for(j=low;j<high-i;j++)
                if(dat[j]>dat[j+1])
                    swap(dat[j],dat[j+1]);
        }
}

template<class T>
void BubbleSort<T>::sort_data()
{
    int low,high;
    dat=this->getd();
    l=this->getl();
    low=0;
    high=l-1;
    bubble_sort(low,high);
    this->setd(dat,l);
}

